ecs-agent/ec2/ec2_metadata_client.go

```diff
@@ -94,21 +97,35 @@ type ec2MetadataClientImpl struct {
 }
 
 // NewEC2MetadataClient creates an ec2metadata client to retrieve metadata
```

Comment:
nit: might be a good idea to add a comment here that the passed in config is optional and if null then we create a new default config. From what we had before, it seems like we always define a new config and then pass it in as a parameter.